### DISCUSSION TITLE
I'm Lost on Reverse Proxy, could really use some help!

### DISCUSSION THREAD

**Discussion by H3is3nb3rg404** [nextcloud/all-in-one#2990] [Questions] [answered]
Hello,
I'm trying to get a reverse proxy setup and I do own a domain. I have two options currently installed on Unraid and Ubuntu 22.04 VM.

I tried Cloudflare and that was a waste of time as I could never get it to work.
I have tried others and can't seem to have any luck.
I have looked into all these options and just not sure which to choose.
https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

Other than what is written in that post is there a good tutorial that works with any purchased domain or something like duckdns.

Thank you to all that can help and what's left of my hair thanks you too!

**Comment by szaimen** [comment, marked as answer]
Hi, can you please post the docker-compose file here that you used and also post the reverse proxy config here that you used? Also are both running on the same server and is the reverse proxy running inside a container or on the host?

**Reply by H3is3nb3rg404** [reply]
I got it for the most part working but not sure this is correct as it **still does not work with the Nextcloud Android app**.
I used this setup from the steps moronmonday526 on Reddit did for vaultwarden but it seemed to work for Nextcloud
https://www.reddit.com/r/selfhosted/comments/oe4dl6/nginx_proxy_manager_getting_internal_error/

1) Go to your profile page on CloudFlare, then API tokens

2) Click Create Token

3) Click "Use template" next to the top option "Edit zone DNS"

4) Under Permissions, click "+Add more"

5) Choose "Zone", "Zone", "Read" from left to right

6) Under Zone Resources, click Select at the far right and choose your domain

7) Change your TTL to be as long as you wish

8) Click Continue to Summary at the bottom

9) Click Create token

10) Click Copy on your API token

11) Switch over to your Nginx Proxy Manager tab in your browser

12) Click Add host

13) Enter your domain name (Note: you must click "Add <domain>" that shows up underneath; don't click out of the field)

14) Under "Forward Hostname" enter the 192.168 IP address of your host and the http (not https) port the service is listening on (Note: I'm running both containers in the same Portainer Stack, so I just entered my VaultWarden container name and port 80)

15) Enable block common exploits

16) Click on the SSL tab

17) Drop down "None" for encryption and choose "Request a new SSL certificate"

18) Enable "Force SSL", "HTTP/2 Support", "HSTS Enabled", and "Use a DNS challenge"

19) Under "DNS Provider", choose CloudFlare

20) Under "Credentials file content", change the token to the token you copied from the CloudFlare page

21) Enter your email at the bottom and agree to the terms

22) Click Save

So I'm not sure if there is a better and easier way to get Nextcloud to work and work with the Nextcloud Android app??

**Reply by Zoey2936** [reply]
just configure the nginx proxy manager as the screenshots of the reverse proxy guide shows it

**Reply by H3is3nb3rg404** [reply]
I did configure nginx and it's online getting a cert from Let's Encrypt.
![image](https://github.com/nextcloud/all-in-one/assets/45325007/72810677-2f9b-461e-8141-3d1e2cac5b09)
![image](https://github.com/nextcloud/all-in-one/assets/45325007/2dfcb78b-6fdf-42f9-be52-e74039b0798f)

However, entering my domain into the Nextcloud app I get the Malformed server configuration error.

**Reply by Zoey2936** [reply]
your screenshots differ with the pictures shown inside the reverse proxy guide

**Reply by H3is3nb3rg404** [reply]
Ok, I changed it and it still works on the web but does not work with the app. I know there must be something I'm missing.

Thank you BTW for the feedback

**Reply by szaimen** [reply]
What exaxtly does not work?

**Reply by H3is3nb3rg404** [reply]
I cannot log into the app, I get the malformed server connection. I do need the app to work.

**Reply by szaimen** [reply]
If you disable `cache assets`, does it work then? 
Also did yiu enter your domain in the app or an ip-address?

**Reply by H3is3nb3rg404** [reply]
cache assets was/is disabled.

I use the domain in the Nextcloud app.

**Reply by szaimen** [reply]
did you enter https://yourncdomain.com ?

**Reply by H3is3nb3rg404** [reply]
Yes

**Reply by szaimen** [reply]
Then I am unfortunately out of ideas. Everything works as expected here so I suspect your networking infrastructure to cause this.

**Reply by H3is3nb3rg404** [reply]
So, ultimately, if I can reach Nexcloud via my domain, there should be no issues reaching it through the App? Being this is going through Cloudflare token, there shouldn't be anything in Cloudflare that would need to change?

**Reply by szaimen** [reply]
> if I can reach Nexcloud via my domain, there should be no issues reaching it through the App?

In theory yes

> Being this is going through Cloudflare token, there shouldn't be anything in Cloudflare that would need to change?

Sorry but I dont know enough about cloudflare to help here.

**Reply by H3is3nb3rg404** [reply]
Is there a better way to do all of this and not necessary using my purchased domain. I do have noip ddns I could use or duckdns. Im just wanting to get the app to work. I have looked through and tried different options in the reverse proxy documentation without much luck. So, if someone has a good tutorial to help, this would be greatly appreciated. Usually, this stuff is not that difficult for me but this Nextcloud AIO is killing me!

**Reply by szaimen** [reply]
Maybw this helps? https://help.nextcloud.com/t/login-flow-errors-and-links-not-opening-macosx-ventura-13-4-and-nextcloud-desktop-v3-8-2/162891/3

**Reply by H3is3nb3rg404** [reply]
I was hopeful on this one but unfortunately, that did not work. Im completely baffled on this and why what appears to be a semi-simple task is so difficult to accomplish. I'm not sure if there is a simpler way other than Cloudflare but I feel like I have tried everything to get a purchased domain to work. I have also tried a free DDNS like duckdns and that did not work for me either.

**Comment by H3is3nb3rg404** [comment]
Here is what I used to get docker compose files up and running
https://gitlab.com/bmcgonag/docker_installs
